File: 0255_Problem_1.rs
// Problem Link: https://leetcode.com/contest/weekly-contest-255/problems/find-greatest-common-divisor-of-array/
// Rust Playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c882a4098ed308973255fe0b059a90a7

pub fn find_gcd(nums: Vec<i32>) -> i32 {
    num::integer::gcd(*nums.iter().max().unwrap(), 
                      *nums.iter().min().unwrap())
}

File: 0255_Problem_1.cpp
// Problem Link: https://leetcode.com/contest/weekly-contest-255/problems/find-greatest-common-divisor-of-array/
// Godbolt: https://godbolt.org/z/zK4zexfza

// C++17
int find_gcd(std::vector<int> const& nums) {
    auto const [a, b] = std::minmax_element(nums.cbegin(), nums.cend());
    return std::gcd(*a, *b);
}

// C++20
int find_gcd(std::vector<int> const& nums) {
    auto const r = std::ranges::minmax_element(nums);
    return std::gcd(*r.min, *r.max);
}
